contracts/cw20-base/src/contract.rs

```diff
@@ -261,6 +299,104 @@ pub fn execute_send(
     Ok(res)
 }
 
+pub fn execute_update_marketing(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    project: Option<String>,
+    description: Option<String>,
+    marketing: Option<String>,
+) -> Result<Response, ContractError> {
+    let mut config = TOKEN_INFO.load(deps.storage)?;
+
+    // If there is no marketing info, there is also noone who can update it, therefore it cannot be
+    // authorised
+    let marketing_info = config
+        .marketing
+        .as_mut()
+        .ok_or(ContractError::Unauthorized {})?;
+
+    if marketing_info
+        .marketing
+        .as_ref()
+        .ok_or(ContractError::Unauthorized {})?
+        != &info.sender
+    {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    match project {
+        Some(empty) if empty.trim().is_empty() => marketing_info.project = None,
+        Some(project) => marketing_info.project = Some(project),
+        None => (),
+    }
+
+    match description {
+        Some(empty) if empty.trim().is_empty() => marketing_info.description = None,
+        Some(description) => marketing_info.description = Some(description),
+        None => (),
+    }
+
+    match marketing {
+        Some(empty) if empty.trim().is_empty() => marketing_info.marketing = None,
```

Comment:
I never saw if statements on the branches of a match.
I had to look twice, but this is an interesting approach.

I would have just matched on Some(marketing) and done the if inside the branch. But this works too and is a bit shorter